src/odemis/acq/move.py

```diff
@@ -0,0 +1,314 @@
+    :param focus: focus driver for optical lens
+    :param rx: (float) rotation movement in x axis
+    :param rz: (float) rotation movement in z axis
+    """
+    try:
+        # Check that the stage X,Y,Z are within the limits
+        for axis in {'x', 'y', 'z'}:
+            if not _isInRange(stage, axis):
+                raise ValueError("Axis %s position is out of active range.", axis)
+        # To hold the ordered sub moves list to perform the tilting/imaging move
+        sub_moves = []
+        # Move focus to FAV_POS_DEACTIVE only if stage rx and rz are equal to 0
+        # Otherwise stop if focus is's not already near FAV_POS_DEACTIVE
+        if util.almost_equal(stage.position.value['rx'], 0, atol=ATOL_POS) and util.almost_equal(
+                stage.position.value['rz'], 0, atol=ATOL_POS):
+            sub_moves.append((focus, focus.getMetadata()[model.MD_FAV_POS_DEACTIVE]))
+        elif not util.almost_equal(focus.position.value, focus.getMetadata()[model.MD_FAV_POS_DEACTIVE],
+                                   atol=ATOL_POS):
+            raise ValueError("Cannot proceed while focus is not near FAV_POS_DEACTIVE position.")
+
+        if rx == 0 and rz == 0:  # Imaging
+            # Get the actual Imaging position (which should be ~ 0 as well)
+            rx = stage.getMetadata()[model.MD_FAV_POS_ACTIVE]['rx']
+            rz = stage.getMetadata()[model.MD_FAV_POS_ACTIVE]['rz']
+            sub_moves.append((stage, {'rz': rz}))
+            sub_moves.append((stage, {'rx': rx}))
+        else:
+            if rx is not None:
+                sub_moves.append((stage, {'rx': rx}))
+            if rz is not None:
+                sub_moves.append((stage, {'rz': rz}))
+
+        if future._task_state == CANCELLED:
+            logging.info("CryoTiltSample procedure cancelled before any sub move.")
+            raise CancelledError()
+
+        for driver, sub_move_dict in sub_moves:
+            run_sub_move(future, driver, sub_move_dict)
+    except CancelledError:
+        logging.info("_doCryoTiltSample cancelled.")
+    except Exception as exp:
+        logging.exception(str(exp))
+        raise
+    finally:
+        with future._task_lock:
+            if future._task_state == CANCELLED:
+                raise CancelledError()
+            future._task_state = FINISHED
+
+
+def run_sub_move(future, driver, sub_move_dict):
+    """
+    Perform the sub moveAbs using the given driver and axis->pos dict
+    :param future: cancellable future of the whole move
+    :param driver: Either the stage of the focus driver
+    :param sub_move_dict: the sub_move axis->pos dict
+    :raises TimeoutError: if the sub move timed out
+    :raises CancelledError: if the sub move is cancelled
+    """
+    try:
+        logging.debug("Performing sub move {} -> {}".format(driver.name, sub_move_dict))
+        future._running_subf = driver.moveAbs(sub_move_dict)
+        future._running_subf.result(timeout=TIMEOUT_DURATION)
```

Comment:
Added a timeout though it's inspired by the autofocus function and may not be needed here.